gatsby-node.ts

```diff
@@ -10,3 +11,110 @@ export const onCreateWebpackConfig: GatsbyNode['onCreateWebpackConfig'] = ({
     },
   });
 };
+
+function checkRelativePaths(obj: any): boolean {
+  const files = [`AgencyInfo.csv`, `DeliverInvest.csv`];
+  for (const edge of obj.data.allFile.edges) {
+    if (!files.includes(edge.node.relativePath)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions;
+
+  const relativePaths = await graphql(
+    `
+      query {
+        allFile(sort: { relativePath: ASC }) {
+          edges {
+            node {
+              relativePath
+            }
+          }
+        }
+      }
+    `,
+  );
+
+  // Check to see if scorecard files exist in the expected location
+  if (!checkRelativePaths(relativePaths)) {
+    throw Error(`Was not able to read in required data`);
+  } else {
+    // Todo: consider destructuring result
+    // Todo: consider adding url path to info file
+    // Todo: see if there's a way to use fragments to break up massive query
+    const result = await graphql(
```

Comment:
There is some stuff in the documentation about not doing all the queries up front like this. I'm thinking this is OK for now?